packages/components-kit/src/components/ComponentForm/ComponentForm.component.js

```diff
@@ -80,6 +82,12 @@ export class TCompForm extends React.Component {
 		this.getMemoizedErrors = memoizeOne(toJS);
 	}
 
+	componentWillReceiveProps(nextProps) {
+		if (this.props.state.get('properties') !== nextProps.state.get('properties')) {
+			this.setState({ properties: nextProps.state.get('properties').toJS() });
```

Comment:
The current use case is not to support a mutation in the state of the properties but more to support edit form.
So i would go to render method with 
`properties: this.state.properties || this.props.state.get('properties').toJS(),`

Reply:
It was to avoid re-computation because of the `.toJS()`. But do you prefer using memoize-one instead of component state ?